src/lesson21/js/userTable.js

```diff
@@ -40,19 +46,113 @@ function makeBodyRow(dataSet) {
   return fragment;
 }
 
-function makeTable(dataSet) {
-  const tableContainer = document.createElement("div");
-  tableContainer.classList.add("table-container");
-  const table = document.createElement("table");
-  table.classList.add("table");
-  const headerData = Object.keys(dataSet.data[0]);
-  makeHeaderRow(headerData);
-  makeBodyRow(dataSet.data);
+const state = ["default", "ascending", "descending"];
+let currentStateIndex = 0;
+let currentState = state[currentStateIndex];
+function changeState() {
+  currentStateIndex = (currentStateIndex + 1) % state.length;
+  currentState = state[currentStateIndex];
+}
+
+function sortData(defaultData) {
+  const copiedData = [...defaultData];
+  switch (currentState) {
+    case "default":
+      return defaultData;
+    case "ascending":
+      return copiedData.sort((a, b) => {
+        return parseInt(a.id) - parseInt(b.id);
+      });
+    case "descending":
+      return copiedData.sort((a, b) => {
+        return parseInt(b.id) - parseInt(a.id);
+      });
+    default:
+      return defaultData;
+  }
+}
+
+function updateBody(data) {
+  document.getElementById("js-tbodyTag").remove();
+  renderTable(makeBodyRow(data));
+}
+
+function updateButtons() {
+  if (document.getElementsByClassName("current-button")[0]) {
+    document
+      .getElementsByClassName("current-button")[0]
+      .classList.remove("current-button");
+  }
+  document
+    .getElementById(`js-${currentState}Button`)
+    .classList.add("current-button");
+}
+
+const buttonsContainer = document.createElement("div");
+buttonsContainer.classList.add("buttons-container");
+function makeSortButton(defaultData) {
+  const buttonsProperty = [
+    {
+      id: "js-defaultButton",
+      state: "default",
+      backgroundImage: "./img/both.svg",
+    },
+    {
+      id: "js-ascendingButton",
+      state: "ascending",
+      backgroundImage: "./img/asc.svg",
+    },
+    {
+      id: "js-descendingButton",
+      state: "descending",
+      backgroundImage: "./img/desc.svg",
+    },
+  ];
+  let sortButtons = [];
+  buttonsProperty.forEach((buttonProperty) => {
+    const sortButton = document.createElement("button");
+    sortButton.id = buttonProperty.id;
+    sortButton.classList.add("sort-button");
+    sortButton.dataset.state = buttonProperty.state;
+    sortButton.style.backgroundImage = buttonProperty.backgroundImage;
```

Comment:
MUST
backgroundの指定ができていないので、svgが表示されていないです。
下記のコードでどうでしょうか？
```suggestion
    sortButton.style.background = `url(${buttonProperty.backgroundImage}) center center/contain no-repeat`;
```

Reply:
> MUST backgroundの指定ができていないので、svgが表示されていないです。 下記のコードでどうでしょうか？

ありがとうございます^^
上手くいきました！！

Reply:
StackBlitz更新されていないかもです...！

Reply:
> StackBlitz更新されていないかもです...！

気づいていませんでした。ありがとうございます^^